example/c/lib_buffer_unpack.c

```diff
@@ -41,6 +41,19 @@ size_t receiver_recv(receiver *r, char* buf, size_t try_size) {
     return actual_size;
 }
 
+size_t receiver_to_unpacker(receiver* r, size_t request_size,
+        msgpack_unpacker *unpacker)
+{
+    // make sure there's enough room, or expand the unpacker accordingly
+    if (msgpack_unpacker_buffer_capacity(unpacker) < request_size) {
+        assert(msgpack_unpacker_reserve_buffer(unpacker, request_size));
```

Comment:
Since msgpack-c already has a `MSGPACK_UNUSED` macro in https://github.com/msgpack/msgpack-c/blob/master/include/msgpack/util.h#L21, it might be a saner approch to simply mark the local variable as "potentially unused":

        bool expanded = msgpack_unpacker_reserve_buffer(unpacker, request_size);
        assert(expanded);
        MSGPACK_UNUSED(expanded);